Crail Kubernetes Deployment on POD Network with revised docker configuration to acomodate Node Discovery

MATRIX4284 opened the pull request:
Crail Kubernetes yaml and Docker build for Automatic Node Discovery

asqasq commented:
Please stop copying files randomly for other repositories.

It seems to me that you don't know what you are doing. Why do you have a hard-coded IP 10.1.47.178? Didyou really try it out? What is file crail-apache/conf/list being used for? Why do you have an endless while loop?

We cannot merge a PR with just randomly copied files.

MATRIX4284 commented:
Hi Adrian,

That code is commented .As i have rewritten the kubernetes to work with
dynamically assigned id from hence i am not using any hardcoded id.
I am using crail-master-service of kubernetes to get
the ips .Pocket does not any kubernetes artifact related to service.I have
created the kubernetes service so that master and slave endpoints are
collected dynamically.
Please check the kubernetes artifact of mine you will notice the difference
in approach.

Thanks,
Kaustav

On Mon, 10 Jun 2019 at 3:41 PM, Adrian Schuepbach <notifications@github.com>
wrote:

> Please stop copying files randomly for other repositories.
>
> It seems to me that you don't know what you are doing. Why do you have a
> hard-coded IP 10.1.47.178? Didyou really try it out? What is file
> crail-apache/conf/list being used for? Why do you have an endless while
> loop?
>
> We cannot merge a PR with just randomly copied files.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-crail/pull/76?email_source=notifications&email_token=AE7YQAHBDCWN2OKILNGZYC3PZYSGZA5CNFSM4HWPT3PKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXJPMEI#issuecomment-500364817>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE7YQAHHOZYJJPMFPSEIKTLPZYSGZANCNFSM4HWPT3PA>
> .
>
-- 
*Thanks & Regards,*
*Kaustav*
*Technical Envagelist*
*Embedded & Distributed Systems*

asqasq commented on kubernetes/dockerfiles/crail-apache/datanode.py:
```diff
@@ -0,0 +1,84 @@
+import socket
+import struct
+
+CONTROLLER_IP = "10.1.47.178"
```
This seems to be a hard-coded IP to me. Why?

MATRIX4284 replied:
Hi i have removed this python as my modules are not using it and with that the hard-coded link too.
Kindly review my changes and suggest me any further changes if any.

MATRIX4284 commented:
Hi Adrian,

I will remove this python file as it is not getting used by my modules .
In fact the new the pull request i created i have removed this file if not
i will remove it asap .
I have created the two artifacts under deploy folder and changed the
crail-site.conf and the slaves where i pointed to my services
The python file you mentioned is not getting used by any of my code.
I can remove it in next update in case i forgot.

Thanks,
Kaustav

Thanks ,
Kaustav

On Mon, 10 Jun 2019 at 3:57 PM, Adrian Schuepbach <notifications@github.com>
wrote:

> *@asqasq* requested changes on this pull request.
> ------------------------------
>
> In kubernetes/dockerfiles/crail-apache/datanode.py
> <https://github.com/apache/incubator-crail/pull/76#discussion_r291958143>:
>
> > @@ -0,0 +1,84 @@
> +##############  Pocket resource utilization daemon  ####################
> +
> +#from jsonsocket import Client
> +import time
> +import psutil
> +import ifcfg
> +import threading
> +import pickle
> +import sys
> +import socket
> +import struct
> +
> +CONTROLLER_IP = "10.1.47.178"
>
> This seems to be a hard-coded IP to me. Why?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-crail/pull/76?email_source=notifications&email_token=AE7YQAGSLAGOOECEMIVJK3TPZYT7PA5CNFSM4HWPT3PKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3AUGWA#pullrequestreview-247546712>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE7YQAHQZ4N6PGM6NWVTRHTPZYT7PANCNFSM4HWPT3PA>
> .
>
-- 
*Thanks & Regards,*
*Kaustav*
*Technical Envagelist*
*Embedded & Distributed Systems*

MATRIX4284 commented:
Hi Adrian,

I will publish the manual shortly once you have a look at my readme you
will understand the flow .Then you will probably understand that my
approach neither has any hard coding nor it is restricted to only aws .

Thanks ,
Kaustav

On Mon, 10 Jun 2019 at 4:07 PM, Kaustav Mukherjee <oracler4284@gmail.com>
wrote:

> Hi Adrian,
>
> I will remove this python file as it is not getting used by my modules .
> In fact the new the pull request i created i have removed this file if not
> i will remove it asap .
> I have created the two artifacts under deploy folder and changed the
> crail-site.conf and the slaves where i pointed to my services
> The python file you mentioned is not getting used by any of my code.
> I can remove it in next update in case i forgot.
>
> Thanks,
> Kaustav
>
> Thanks ,
> Kaustav
>
> On Mon, 10 Jun 2019 at 3:57 PM, Adrian Schuepbach <
> notifications@github.com> wrote:
>
>> *@asqasq* requested changes on this pull request.
>> ------------------------------
>>
>> In kubernetes/dockerfiles/crail-apache/datanode.py
>> <https://github.com/apache/incubator-crail/pull/76#discussion_r291958143>
>> :
>>
>> > @@ -0,0 +1,84 @@
>> +##############  Pocket resource utilization daemon  ####################
>> +
>> +#from jsonsocket import Client
>> +import time
>> +import psutil
>> +import ifcfg
>> +import threading
>> +import pickle
>> +import sys
>> +import socket
>> +import struct
>> +
>> +CONTROLLER_IP = "10.1.47.178"
>>
>> This seems to be a hard-coded IP to me. Why?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/apache/incubator-crail/pull/76?email_source=notifications&email_token=AE7YQAGSLAGOOECEMIVJK3TPZYT7PA5CNFSM4HWPT3PKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3AUGWA#pullrequestreview-247546712>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AE7YQAHQZ4N6PGM6NWVTRHTPZYT7PANCNFSM4HWPT3PA>
>> .
>>
> --
> *Thanks & Regards,*
> *Kaustav*
> *Technical Envagelist*
> *Embedded & Distributed Systems*
>
-- 
*Thanks & Regards,*
*Kaustav*
*Technical Envagelist*
*Embedded & Distributed Systems*

asqasq commented:
Hi Kaustav

Actually I do understand the flow.

This PR cannot be merged for several reasons:
1. In my opinion it is not the right way to just copy files from other people and claim that
   they are youres (even more, you copied our files from our repositories)
2. The Pocket files do not work with the latest Crail version. Pocket was a research project
    based on a older and especially a modified version of Crail.
3. It seems to me that you don't understand what you are doing
4. I doubt that you tested it, especially after your last change. Deleting few files
    could potentially mean that something does not work anymore (I doubt that it worked before).
5. We cannot reference docker images from a non-trusted source in the yaml files.

Therefore, we cannot merge this PR.

MATRIX4284 commented:
Hi Adrian,

Thanks for your feedback .
I will close this PR.

Thanks
Kaustav

On Tue, 11 Jun 2019 at 1:16 PM, Adrian Schuepbach <notifications@github.com>
wrote:

> Hi Kaustav
>
> Actually I do understand the flow.
>
> This PR cannot be merged for several reasons:
>
>    1. In my opinion it is not the right way to just copy files from other
>    people and claim that
>    they are youres (even more, you copied our files from our repositories)
>    2. The Pocket files do not work with the latest Crail version. Pocket
>    was a research project
>    based on a older and especially a modified version of Crail.
>    3. It seems to me that you don't understand what you are doing
>    4. I doubt that you tested it, especially after your last change.
>    Deleting few files
>    could potentially mean that something does not work anymore (I doubt
>    that it worked before).
>    5. We cannot reference docker images from a non-trusted source in the
>    yaml files.
>
> Therefore, we cannot merge this PR.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-crail/pull/76?email_source=notifications&email_token=AE7YQABGGZHOKE5HDIUZ7ILPZ5J5VA5CNFSM4HWPT3PKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXMH44A#issuecomment-500727408>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE7YQAG3BOOBPOCJTF4H65DPZ5J5VANCNFSM4HWPT3PA>
> .
>
-- 
*Thanks & Regards,*
*Kaustav*
*Technical Envagelist*
*Embedded & Distributed Systems*

MATRIX4284 commented:
Closing the PR

MATRIX4284 commented:
..
